src/displayapp/screens/FileView.cpp

```diff
@@ -0,0 +1,128 @@
+#include <nrf_log.h>
+#include "displayapp/screens/FileView.h"
+#include "displayapp/DisplayApp.h"
+#include "displayapp/InfiniTimeTheme.h"
+
+using namespace Pinetime::Applications::Screens;
+
+FileView::FileView(uint8_t screenID, uint8_t nScreens, DisplayApp* app, const char* path)
+  : screenID(screenID), nScreens(nScreens), Screen(app) {
+  label = nullptr;
+  pageIndicatorBase = nullptr;
+  pageIndicator = nullptr;
+
+  const char* c = strrchr(path, '/') + 1;
+  if (c == nullptr)
+    c = path;
+
+  strncpy(name, c, LFS_NAME_MAX - 1);
+  char* pchar = strchr(name, '_');
+  while (pchar != nullptr) {
+    *pchar = ' ';
+    pchar = strchr(pchar + 1, '_');
+  }
+
+  pageIndicatorBasePoints[0].x = 0;
```

Comment:
You are copying `displayapp/widgets/PageIndicator.cpp` here, right? Can you use it instead? Or can you modify it reasonably to use it?